File: app/src/main/java/com/chooloo/www/callmanager/ui/base/BasePresenter.java
package com.chooloo.www.callmanager.ui.base;

import android.content.Context;

import androidx.annotation.NonNull;
import androidx.lifecycle.Lifecycle;
import androidx.lifecycle.LifecycleObserver;

public class BasePresenter<V extends MvpView> implements MvpPresenter<V>, LifecycleObserver {

    private static final String TAG = "BasePresenter";

    protected V mMvpView;

    @Override
    public void onAttach(V mvpView) {
        mMvpView = mvpView;
    }

    @Override
    public void onDetach() {
        mMvpView = null;
    }
}

File: app/src/main/java/com/chooloo/www/callmanager/ui/base/BasePresenter.kt
package com.chooloo.www.callmanager.ui.base

import androidx.lifecycle.LifecycleObserver

open class BasePresenter<V : MvpView?> : MvpPresenter<V>, LifecycleObserver {
    @JvmField protected var mMvpView: V? = null

    override fun attach(mvpView: V) {
        mMvpView = mvpView
    }

    override fun detach() {
        mMvpView = null
    }
}